### ISSUE TITLE
Issue with windows installation: `NoneType` object has no attribute `data`

### ISSUE THREAD

**Issue by laitcl** [ReactionMechanismGenerator/RMG-Py#880]
There seems to be a reoccurring issue with some windows installations.

```
Traceback (most recent call last):
  File "rmg.py", line 147, in <module>
    cProfile.runctx(command, global_vars, local_vars, stats_file)
  File "C:\Anaconda\envs\rmg_env\lib\cProfile.py", line 49, in runctx
    prof = prof.runctx(statement, globals, locals)
  File "C:\Anaconda\envs\rmg_env\lib\cProfile.py", line 140, in runctx
    exec cmd in globals, locals
  File "<string>", line 1, in <module>
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\m
ain.py", line 529, in execute
    self.initialize(**kwargs)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\m
ain.py", line 428, in initialize
    spec, isNew = self.reactionModel.makeNewSpecies(molecule, label=label, react
ive=False)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\m
odel.py", line 330, in makeNewSpecies
    submit(spec)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\therm
o\thermoengine.py", line 136, in submit
    spc.thermo = submit_(evaluator, spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\scoop
_framework\util.py", line 171, in submit_
    return func(*args, **kwargs)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\therm
o\thermoengine.py", line 122, in evaluator
    thermo = generateThermoData(spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\therm
o\thermoengine.py", line 103, in generateThermoData
    thermo0 = thermodb.getThermoData(spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\
thermo.py", line 1096, in getThermoData
    thermo0 = self.getThermoDataFromGroups(species)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\
thermo.py", line 1242, in getThermoDataFromGroups
    tdata = self.estimateThermoViaGroupAdditivity(molecule)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\
thermo.py", line 1374, in estimateThermoViaGroupAdditivity
    thermoData = self.computeGroupAdditivityThermo(molecule)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\
thermo.py", line 1413, in computeGroupAdditivityThermo
    self.__addGroupThermoData(thermoData, self.groups['group'], molecule, {'*':a
tom})
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\
thermo.py", line 1621, in __addGroupThermoData
    while node.data is None and node is not None:
AttributeError: 'NoneType' object has no attribute 'data'
```

**Comment by alongd** [comment]
Looks like RMG couldn't find thermo data. Did you receive this error after running a certain job? (If so, could you attach the input file?)

**Comment by laitcl** [comment]
I received this error running make eg1

**Comment by mliu49** [comment]
Sorry, I posted this issue from @laitcl's computer, but didn't have a chance to post a follow-up comment until now.

This is the third time I've come across this error. All three users followed the instructions for installation by source using Anaconda for windows.

In two of the three cases, they had a working installation of RMG, and then uninstalled everything (including Anaconda) and reinstalled. The error appeared after reinstalling. The other case was a new installation.

It seems that the logic for the while statement itself is not the best, but it's not problematic for unix installations or most windows installations, so something else must be wrong.

Recompiling didn't help. The system PATH variable was checked, and removing unnecessary items from the path didn't help. PYTHONPATH also looked normal.

Unfortunately, I'm not sure of a straightforward way to reproduce the problem on a different computer.

**Comment by rwest** [comment]
This may help:
https://github.com/rwest/RMG-Py/commit/1307d4b6bef9e3d02282fbb9041664e526433de4
but no idea if that's just fixing a symptom, or fixing one but not both of the problems. It certainly doesn't explain why it's sporadic...

(I have a few commits like that sitting on our RMG-cat branch that I mean to back-port into a pull request for master... one day)

**Comment by KEHANG** [comment]
The next line [#1321](https://github.com/rwest/RMG-Py/commit/1307d4b6bef9e3d02282fbb9041664e526433de4#diff-62923869fe80e33ca1b9311680f66922R1321) will still catch the case where`node == None` and raise an error. 

By reading the method of `__addGroupThermoData`, it seems that parent (or grandparent) of `node0` is `None`, which is odd (maybe the database is not correctly loaded?). Would be helpful if you can print out which species is causing the descending tree problem.

**Comment by rwest** [comment]
Yes, my patch would just avoid the confusing stack trace and give you a slightly more helpful error message.

**Comment by laitcl** [comment]
With @rwest's patch, the error message changed to the following, which is as @KEHANG predicted.
```
Traceback (most recent call last):
  File "rmg.py", line 147, in <module>
    cProfile.runctx(command, global_vars, local_vars, stats_file)
  File "C:\Anaconda\envs\rmg_env\lib\cProfile.py", line 49, in runctx
    prof = prof.runctx(statement, globals, locals)
  File "C:\Anaconda\envs\rmg_env\lib\cProfile.py", line 140, in runctx
    exec cmd in globals, locals
  File "<string>", line 1, in <module>
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\main.py", line 529, in execute
    self.initialize(**kwargs)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\main.py", line 428, in initialize
    spec, isNew = self.reactionModel.makeNewSpecies(molecule, label=label, reactive=False)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\rmg\model.py", line 330, in makeNewSpecies
    submit(spec)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\thermo\thermoengine.py", line 136, in submit
    spc.thermo = submit_(evaluator, spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\scoop_framework\util.py", line 171, in submit_
    return func(*args, **kwargs)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\thermo\thermoengine.py", line 122, in evaluator
    thermo = generateThermoData(spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\thermo\thermoengine.py", line 103, in generateThermoData
    thermo0 = thermodb.getThermoData(spc)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\thermo.py", line 1096, in getThermoData
    thermo0 = self.getThermoDataFromGroups(species)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\thermo.py", line 1242, in getThermoDataFromGroups
    tdata = self.estimateThermoViaGroupAdditivity(molecule)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\thermo.py", line 1374, in estimateThermoViaGroupAdditivity
    thermoData = self.computeGroupAdditivityThermo(molecule)
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\thermo.py", line 1413, in computeGroupAdditivityThermo
    self.__addGroupThermoData(thermoData, self.groups['group'], molecule, {'*':atom})
  File "C:\Users\User1\Desktop\Misc\Green_Group\RMG_Workspace\RMG-Py\rmgpy\data\thermo.py", line 1624, in __addGroupThermoData
    raise DatabaseError('Unable to determine thermo parameters for {0}: no data for node {1} or any of its ancestors.'.format(molecule, node0) )
DatabaseError: Unable to determine thermo parameters for <Molecule "[Ar]">: no data for node R or any of its ancestors.
```

**Comment by mliu49** [comment]
This makes it seem as if primaryThermoLibrary is not being checked, since that is where the thermo for [Ar] should come from. The log does say that it is being loaded though.

**Comment by KEHANG** [comment]
I was able to reproduce the error recently. My hypothesis is `openbabel` is not functioning on your machine. If so, `rdkit` will be used to create molecules. In the `parser.py`, we have
```python
SMILES_LOOKUPS = {
            '[He]':# RDKit improperly handles helium and returns it in a triplet state
            """
            He
            multiplicity 1
            1 He u0 p1
            """
}   
```
to help `rdkit` parse `He` SMILES. But unfortunately `rdkit` cannot deal with `Ar` either. If we extend the dictionary by adding `Ar`, the error will be gone.

```python
SMILES_LOOKUPS = {
            '[He]':# RDKit improperly handles helium and returns it in a triplet state
            """
            He
            multiplicity 1
            1 He u0 p1
            """,
            '[Ar]':# RDKit improperly handles argon 
            """
            Ar
            multiplicity 1
            1 Ar u0 p1
            """
}   
```
I guess the question is why `openbabel` binary doesn't work for some windows machines. Seems it's not because of different versions of windows. Souyma's windows 10 can run openbabel and have no such issue and some windows 10 (including mine) does have the problem.

@laitcl if you just want to make the rmg start running again, one solution I've tested out is to use `openbabel` binary from other channels (e.g., `openbabel`, `hjkgroup`). Their binaries can eliminate the problem.

But one thing I still don't understand is `rmg` channel's openbabel has been there for a while and seems working all the time previously. I'll investigate a little bit more. If anyone has any ideas triggered by this new piece of information, let me know.

**Comment by rwest** [comment]
I know this issue was originally about windows installation, but I'm hijacking it briefly to ask @KEHANG are you sure about that adjacency list for Argon?  When I parse the '[Ar]' I get 4 lone pairs. When I parse your adjlist (with 1 lone pair) I get `InvalidAdjacencyListError: Invalid valency for atom Ar (Ar) with 0 unpaired electrons, 1 pairs of electrons, 0 charge, and bonds [].`

Should it be this?
```
            Ar
            multiplicity 1
            1 Ar u0 p4
```
(I'm working on #946)

**Comment by KEHANG** [comment]
@rwest you're right! It's a typo.

**Comment by mliu49** [comment]
Should be fixed by #949 or by reinstalling openbabel.
